app/models/game.rb

```diff
@@ -32,7 +32,8 @@ def check(id, opponent_id)
 
   def determine_checkmate
     return checkmate(white_player_id) if turn_number.even? && determine_check
-    checkmate(black_player_id) if turn_number.odd? && determine_check
+    return checkmate(black_player_id) if turn_number.odd? && determine_check
+    false
```

Comment:
I'm curious why you changed this.  Since `checkmate` returns either `true` or `false`, calling `checkmate` in `determine_checkmate` will return whatever `checkmate` returns. By doing this, you're just ignoring that `checkmate` already returns false.